### ISSUE TITLE
Your ONS API function

### ISSUE THREAD

**Issue by francisbarton** [dfe-analytical-services/dfeR#85]
Hi there,

I was pointed to your ONS API functions today, nice work.
I just wanted to let you know that I've had a package available for a few years now that does some very similar things. In case it's useful to you or provides any functionality you don't yet have.

[boundr](https://github.com/francisbarton/boundr)
[pkgdown site](https://francisbarton.github.io/boundr/)

I'd be interested in any feedback you have for me on the UI and making the functions user-friendly. Your shorthand functions are a neat idea.
My main aim was to avoid the user having to go to the OG site and look up the table name they needed.

Please see what you think, and maybe let me know any ideas! Thanks.

**Comment by francisbarton** [comment]
Your

```r
fetch_las(2022, c("Scotland", "Northern Ireland"))
```

would be my:

```r
boundr::lookup("lad", "ctry", c("Scotland", "Northern Ireland"), lookup_yea 
     r = 2022)
```

**Comment by cjrace** [comment]
Very nice! We had no idea this existed so I'm very happy you've raised this issue - I think there'll be things we might have expanded `get_ons_api_data()` to do that we don't need to anymore (and also I'm excited to block out some time to dig through your source code to understand how you approached it!) - thank you for sharing 😄 

Our `fetch_` functions are more bespoke to how we publish the data (such as England LAs, where we take UTLA and then adapt it slightly for metropolitan areas, and combining Inner / Outer London counties into the regions table, as well as adding some custom locations in on top of that too). All of our `fetch_` functions pull from data we've curated and stashed in the package itself. These two for example differ quite a lot as the dfeR version has the data ready to go and can give the data frame instantly and it also gives different locations due to how we've curated it:

```
boundr::lookup("utla", "ctry", c("England"), lookup_year = 2022)
dfeR::fetch_las(2022, "England")
```

I think my first thought for boundr is that it might be nice to have an easy reference table for the shorthands of levels that are available (or at least most common ones), that way you can really avoid needing to go onto the Open Geography Portal, as you could scan that and use the values in that to build your arguments for the `boundr::lookup()` function. I made our shorthand purely for my own purposes behind the scenes tidying up the data set names, though I think being able to see what options I could feed into `lookup()` would be really helpful for when we make our own versions of other data sets.
 
What I'd suggest from our side to respond to this and be able to mark this issue as closed is that:
1. We add a reference to your package in our README
2. We add a reference in our documentation of the `get_ons_api_data()` function
3. We make a note in our contributing file for future dfeR contributors so they're aware they can make use of boundr when trying to create new datasets moving forwards (I'd expect we can use it quite a lot in our data-raw/ scripts)

These changes on our side will help make sure we can use your package instead of extending our function to avoid duplicating, which will also mean we can pass on any things to you that we come up with that might be helpful to add into boundr, so hopefully a win-win!

**Comment by francisbarton** [comment]
Thanks Cam, appreciate the positive reponse.

It would be neat to have a reference to my package within your docs as for some cases where you don't already have the data built-in and ready to go it might be useful to your colleagues.

And yes you're right about documenting the shorthands more thoroughly - it's [on my list](https://codeberg.org/francisbarton/boundr/issues/9)! (I did actually have a version of this in my README a little while ago but took it out because I was refactoring some related things).

Once that is done I am considering making v0.4 a release and submitting it to CRAN. I think some of my error messages could be better too.

Here's to the win-win.

Hopefully some of my behind-the-scenes functions for querying the API endpoint might be interesting - I've got some systems in place for batching up requests and iterating over the batches to try to avoid hitting the rate limits etc.
